provider_shopper/lib/models/catalog.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+
+/// A proxy of the catalog of items the user can buy.
+///
+/// In a real app, this might be backed by a backend and cached on device.
+/// In this sample app, the catalog is procedurally generated.
+///
+/// For simplicity, the catalog is expected to be immutable (no products are
+/// expected to be added, removed or changed during the execution of the app).
+class CatalogModel {
+  static const _itemNames = [
+    'Code Smell',
+    'Control Flow',
+    'Interpreter',
+    'Recursion',
+    'Sprint',
+    'Heisenbug',
+    'Spaghetti',
+    'Hydra Code',
+    'Off-By-One',
+    'Scope',
+    'Callback',
+    'Closure',
+    'Automata',
+    'Bit Shift',
+    'Currying',
+  ];
+
+  /// Get item by [id].
+  Item getById(int id) => Item(id, _itemNames[id % _itemNames.length]);
+
+  /// Get item by its position in the catalog.
+  Item getByPosition(int index) {
+    // In our simplified case, an item's position in the catalog is also its id.
+    return getById(index);
```

Comment:
I've thought about it, but I think rewriting would make the matter even worse than it is. We would still need to explain why we're using the same code as above, otherwise it looks like a copy-paste bug.

```dart
  /// Get item by [id].
  Item getById(int id) => Item(id, _itemNames[id % _itemNames.length]);

  /// Get item by [position].
  Item getByPosition(int position) => Item(position, _itemNames[position % _itemNames.length]);
```

 And if we explain, it's better to just call the other method. I don't feel strongly about this, though, so feel free to push back.